ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/systemcall/SystemCallProcessor.java

```diff
@@ -48,37 +49,33 @@ public SystemCallProcessor(final MainnetTransactionProcessor mainnetTransactionP
   }
 
   /**
-   * Processes a system call to a specified address, using the provided world state, block header,
-   * operation tracer, and block hash lookup.
+   * Processes a system call.
    *
-   * @param callAddress the address to call.
-   * @param worldState the current world state.
-   * @param blockHeader the current block header.
-   * @param operationTracer the operation tracer for tracing EVM operations.
-   * @param blockHashLookup the block hash lookup function.
-   * @return the output data from the call. If no code exists at the callAddress then an empty Bytes
-   *     is returned.
+   * @param callAddress The address to call.
+   * @param context The system call context. The input data to the system call.
+   * @return The output of the system call.
    */
   public Bytes process(
-      final Address callAddress,
-      final WorldUpdater worldState,
-      final ProcessableBlockHeader blockHeader,
-      final OperationTracer operationTracer,
-      final BlockHashLookup blockHashLookup) {
+      final Address callAddress, final BlockProcessingContext context, final Bytes inputData) {
+    WorldUpdater updater = context.getWorldState().updater();
```

Comment:
for more visibility I would have liked to see the commit of the updater at the same level as the creation of this one. Because we have to go down in the levels to understand that we are committing the updater in certain conditions

Reply:
Cool! Done!